src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -6956,16 +6956,27 @@ and TcObjectExpr (cenv: cenv) env tpenv (objTy, realObjTy, argopt, binds, extraI
 //-------------------------------------------------------------------------
 
 /// Check a constant string expression. It might be a 'printf' format string
-and TcConstStringExpr cenv (overallTy: OverallTy) env m tpenv (s: string) =
-
-    let g = cenv.g
+and TcConstStringExpr cenv (overallTy: OverallTy) env m tpenv (s: string) isInlineLiteral =
```

Comment:
Can we have a named DU instead of boolean flag here?

type LiteralType = Inline | StaticField